x-pack/test_serverless/api_integration/test_suites/common/spaces.ts

```diff
@@ -13,24 +13,206 @@ export default function ({ getService }: FtrProviderContext) {
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [get] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .get('/api/spaces/space/default')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect succes because we're using the internal header
+          expect(body).toEqual(
+            expect.objectContaining({
+              id: 'default',
+            })
+          );
+          expect(status).toBe(200);
+        });
+
+        it('#getAll requires internal header', async () => {
+          let body: any;
+          let status: number;
+
+          ({ body, status } = await supertest
+            .get('/api/spaces/space')
+            .set(svlCommonApi.getCommonRequestHeader()));
+          // expect a rejection because we're not using the internal header
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [get] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .get('/api/spaces/space')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect succes because we're using the internal header
+          expect(body).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                id: 'default',
+              }),
+            ])
+          );
+          expect(status).toBe(200);
+        });
+
+        it('#getActiveSpace requires internal header', async () => {
+          let body: any;
+          let status: number;
+
+          ({ body, status } = await supertest
+            .get('/internal/spaces/_active_space')
+            .set(svlCommonApi.getCommonRequestHeader()));
+          // expect a rejection because we're not using the internal header
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [get] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .get('/internal/spaces/_active_space')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect succes because we're using the internal header
+          expect(body).toEqual(
+            expect.objectContaining({
+              id: 'default',
+            })
+          );
+          expect(status).toBe(200);
+        });
+
+        it('#update requires internal header', async () => {
```

Comment:
@thomheymann since we're hiding Spaces selector in the Serverless MVP, is there any other reason why we might want to keep Spaces Update API enabled in Serverless?

Reply:
The only thought I had was if the default space should be configurable with a different name for different project types. But I don't think that is a requirement/ask.

Reply:
> But I don't think that is a requirement/ask.

Yeah, that's my understanding too.

Reply:
The only reason I can think of is to ensure that the manage spaces screen remains functional. If we hide that screen we can also disable the update spaces API.